uiTests/src/test/resources/scenarios-and-utils.rhino.js

```diff
@@ -72,21 +74,48 @@ function openTabAndReturnRowCount(project) {
     toolWindow.activate(function () {});
   });
 
-  let graphTable;
   GuiUtils.runOrInvokeAndWait(function () {
     const contentManager = toolWindow.getContentManager();
     const tab = contentManager.findContent('Git Machete');
     contentManager.setSelectedContent(tab);
-    const panel = tab.getComponent();
-    graphTable = panel.getGraphTable();
-    graphTable.queueRepositoryUpdateAndModelRefresh();
   });
+}
 
-  let graphTableRowCount;
+function _getGraphTable(project) {
+  const toolWindowManager = ToolWindowManager.getInstance(project);
+  const toolWindow = toolWindowManager.getToolWindow(ToolWindowId.VCS);
+  const contentManager = toolWindow.getContentManager();
+  const tab = contentManager.findContent('Git Machete');
+  const panel = tab.getComponent();
+  return panel.getGraphTable();
+}
+
+function _refreshModelAndWaitUntilDone(graphTable) {
+  let refreshDone = false;
+  graphTable.queueRepositoryUpdateAndModelRefresh(/* doOnUIThreadWhenReady */ function() {
+    refreshDone = true;
```

Comment:
Nice, not doable directly in Java (possible since in JS the entire lexical scope is passed to the closure)